scripts/http-cookie-alert.nse

```diff
@@ -0,0 +1,102 @@
+local http = require "http"
+local shortport = require "shortport"
+local stdnse = require "stdnse"
+local string = require "string"
+local httpspider = require "httpspider"
+local table = require "table"
+local httpcookies = require "httpcookies"
+local nmap = require "nmap"
+
+description = [[
+The script allows users to quickly
+check if web applications return any interesting cookie names/values and informs the user
+regarding the same.
+
+The cookie name (or value) must be meaningless to prevent information disclosure attacks,
+where an attacker is able to decode the contents of the ID and extract details of the user,
+the session, or the inner workings of the web application. The script calls the httpspider 
+library with docookies argument enabled and it received all the cookies in a cookiejar. It 
+then searches the cookiejar received for any interesting values.
+
+]]	
+
+---
+-- @usage
+-- nmap -p 80 --script http-cookie-alert <target>
+--
+-- @output
+-- PORT   STATE SERVICE REASON
+-- 80/tcp open  http    syn-ack
+-- | http-cookie-alert: 
+-- | The following interesting cookies were found: 
+-- |_PHPSESSIONID	!o/BcH3Rvj1iaqD7220J9/DO4/a/fT0IotERS7sHae2NOCOCXAS/oDJPToaKcAy+/LB+Erqc3EQwt
+-- 
+--
+-- @xmloutput
+-- <table key="interesting-cookies">
+-- <elem>/rest/contactsjp.php</elem>
+-- </table>
+-- 
+-- @args http-cookie-alert The URL path to request. The default path is "/".
+--- 
+
+local sensitive = {"sessionid", "token", "admin"}
+
+author = {"Vinamra Bhatia"}
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"safe", "vuln", "discovery"}
+
+portrule = shortport.http
+
+action = function(host, port)
+  local path = stdnse.get_script_args(SCRIPT_NAME .. ".path") or "/"
+  local output_xml = stdnse.output_table()
+  output_xml = {}
+  output_xml['interesting-cookies'] = {}
+  local output_str = "\nThe following interesting cookies were found: " 
+
+  -- crawl to put the cookies received in various web pages.
+  local crawler = httpspider.Crawler:new(host, port, path, {scriptname = SCRIPT_NAME})
+
+  if (not(crawler)) then
+    return
+  end
+
+  crawler:set_docookies(true)
```

Comment:
A better name for 'docookies' is 'enable_cookies'. Please update everything accordingly.

Reply:
This should be changed to the new method name to reflect your latest changes.